sentry-core/src/main/java/io/sentry/core/hints/CachedEnvelopeHint.java

```diff
@@ -0,0 +1,65 @@
+package io.sentry.core.hints;
+
+import static io.sentry.core.SentryLevel.ERROR;
+
+import io.sentry.core.ILogger;
+import io.sentry.core.util.Objects;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+@ApiStatus.Internal
+public final class CachedEnvelopeHint
+    implements Cached, Retryable, SubmissionResult, Flushable, Resettable, ApplyScopeData {
```

Comment:
Hints before were two private classes implementing 3 interfaces.
I believe it got a lot more complex right now. Public classes, this one implementing 7 interfaces?

Is this whole refactor needed? If NDK events didn't get Scope applied can we find a simpler way? Like didn't the `Cached` hint alone control that?
What did `Cached` control? Maybe we can replace it with something explicit as u added: `ApplyScopeData` if that was the sole job of `Cached`?

Reply:
they are not private because they are organized in different packages, but there's a `@ApiStatus.Internal` annotation.

as this class doesn't belong to the class anymore, `interfaces` were needed to call respective methods like `waitFlush`.

Only the `Cached` hint was not possible to fix the issue if not changing way more things.

`Cached` is also used for storing files on disk again or not.

Reply:
> Cached is also used for storing files on disk again or not.

Right, I see.
So seems like we need all of them except `Resettable`, right?